src/Rings/mpoly-graded.jl

```diff
@@ -2424,6 +2424,12 @@ function AbstractAlgebra.promote_rule(::Type{MPolyDecRingElem{S, T}}, ::Type{U})
   end
 end
 
+# For some reason the following is necessary to remove ambiguities
+function AbstractAlgebra.promote_rule(::Type{MPolyDecRingElem{S, T}}, ::Type{MPolyDecRingElem{S, T}}) where {S, T}
+  return MPolyDecRingElem{S, T}
+end
```

Comment:
Unfortunately, I didn't write down a minimal example for this. But if you too think this is weird and would like to investigate this, remove the method and run the scheme's test code.

Reply:
This is fine.